compiler/x/amd64/objectfmt/OMRJitCodeRXObjectFormat.cpp

```diff
@@ -0,0 +1,187 @@
+      }
+
+   ccFunctionDataAddress->address = targetAddress;
+
+   TR::StaticSymbol *functionDataSymbol =
+      TR::StaticSymbol::createWithAddress(comp->trHeapMemory(), TR::Address, reinterpret_cast<void *>(ccFunctionDataAddress));
+   functionDataSymbol->setNotDataAddress();
+   TR::SymbolReference *functionDataSymRef = new (comp->trHeapMemory()) TR::SymbolReference(comp->getSymRefTab(), functionDataSymbol, 0);
+
+   TR::X86CallMemInstruction *callInstr;
+   TR::MemoryReference *callMR = new (comp->trHeapMemory()) TR::MemoryReference(functionDataSymRef, data.cg, true);
+
+   if (data.prevInstr)
+      {
+      callInstr = generateCallMemInstruction(data.prevInstr, CALLMem, callMR, data.regDeps, data.cg);
+      }
+   else
+      {
+      callInstr = generateCallMemInstruction(CALLMem, data.callNode, callMR, data.regDeps, data.cg);
+      }
+
+   if (data.adjustsFramePointerBy != 0)
+      {
+      callInstr->setAdjustsFramePointerBy(data.adjustsFramePointerBy);
+      }
+
+   data.out_callInstr = callInstr;
+
+   return callInstr;
+   }
+
+
+uint8_t *
+OMR::X86::AMD64::JitCodeRXObjectFormat::encodeFunctionCall(TR::FunctionCallData &data)
+   {
+   TR::SymbolReference *methodSymRef;
+
+   if (data.runtimeHelperIndex > 0)
+      {
+      methodSymRef = data.cg->symRefTab()->findOrCreateRuntimeHelper(data.runtimeHelperIndex);
+      }
+   else
+      {
+      methodSymRef = data.methodSymRef;
+      }
+
+   /**
+    * If a targetAddress has been provided then use that to override whatever would be
+    * derived from the symbol reference.
+    */
+   TR_ASSERT_FATAL(!(data.runtimeHelperIndex && data.targetAddress), "a runtime helper and target address cannot both be provided");
+
+   uintptr_t targetAddress = data.targetAddress ? data.targetAddress : (uintptr_t)methodSymRef->getMethodAddress();
+
+   TR_ASSERT_FATAL(targetAddress, "function address is unknown");
+
+   data.cg->resetIsLeafMethod();
+
+   TR::Compilation *comp = data.cg->comp();
+
+   ccFunctionData *ccFunctionDataAddress =
+      reinterpret_cast<ccFunctionData *>(data.cg->allocateCodeMemory(sizeof(ccFunctionData), false));
+
+   if (!ccFunctionDataAddress)
+      {
+      comp->failCompilation<TR::CompilationException>("Could not allocate function data");
+      }
+
+   ccFunctionDataAddress->address = targetAddress;
+
+   *data.bufferAddress++ = 0xff;  // CALL | JMP Mem
+   uint8_t modRM = data.isCall ? 0x15 : 0x25;  // RIP addressing mode
+   *data.bufferAddress++ = modRM;
+
+   TR_ASSERT_FATAL(comp->target().cpu.isTargetWithinRIPRange(
+                      reinterpret_cast<intptr_t>(ccFunctionDataAddress),
+                      reinterpret_cast<intptr_t>(data.bufferAddress+4)),
+                      "ccFunctionData must be reachable directly");
+
+   int32_t disp32 = (int32_t)(reinterpret_cast<intptr_t>(ccFunctionDataAddress) - reinterpret_cast<intptr_t>(data.bufferAddress+4));
+   *(int32_t *)data.bufferAddress = disp32;
```

Comment:
`reinterpret_cast` need here as well as elsewhere. Will stop commenting on this one.